src/apps/common/submodules/numbers/numbers.js

```diff
@@ -811,6 +825,47 @@ define(function(require) {
 				});
 			};
 
+			var searchExternalNumber = function(searchString, parent) {
```

Comment:
As the sub-accounts' numbers will no longer be pre-loaded, this needs to be refactored. Check how the function `searchListNumbers`  that is a few lines above works and use it as an example, or if possible use that same function, as it should work in a similar way for the 3 number tabs.